mlflow/store/artifact_repo.py

```diff
@@ -43,7 +43,9 @@ def log_artifacts(self, local_dir, artifact_path=None):
     @abstractmethod
     def list_artifacts(self, path):
         """
-        Return all the artifacts for this run_uuid directly under path.
+        Return all the artifacts for this run_uuid directly under path. If path is a file, returns
```

Comment:
From offline discussion with @aarondav - returning an empty list when listing a file isn't super intuitive. Ideally we'd either error (Java behavior) or return the file itself (Linux). However, we'll defer that change as a follow-up in favor of documenting the current behavior here to get this fix in.

Reply:
Let's include that this will return an error in the case where the path is neither a file nor directory. We don't have to specify what kind of error...